docs: app mode and cluster mode explained (#1041)
spectrocloud/librarium: content/docs/01-introduction/05-palette-modes.md

App Mode is a unique experience enabled by Palette that aims to remove Kubernetes infrastructure overhead as much as possible. In app mode, you, as the consumer, instead focus on creating and managing App Profiles. App profiles are declarative templates that you can use to define all the required services, containers, and databases that make up an application. Once an app profile is defined, you can deploy the application to any Palette Virtual Cluster by specifying the respective app profile.

App mode comes with an out-of-the-box cluster group managed by us here at Spectro Cloud called beehive.  This cluster group, which under the cover is a collection of Kubernetes clusters, is configured to support Palette Virtual Clusters. As a consumer, you can deploy a new virtual cluster to the beehive cluster group and get started with a Kubernetes cluster in minutes. 

The ability to get started with a Kubernetes cluster in minutes is what makes app mode a powerful development tool. You can use the virtual clusters in an ephemeral manner, such as testing, ad-hoc development, or any other scenario where you would want a short-lived Kubernetes environment up and running in minutes.  

> Alternatively, you could also use App Mode to offer your own Palette-managed host clusters as a PaaS experience to downstream consumers. This concept is easier explained through an example. Assume you are a system administrator, and you want to expose Kubernetes to various in-house development teams. You could deploy several Kubernetes clusters to various platforms and create a " development " cluster group. You also ensured every cluster is enabled for Palette Virtual Cluster by selecting the option before deployment. You can now direct your organization members to use app mode and create Palette Virtual Clusters as needed, or you can create virtual clusters ahead of time for them. The organization members or downstream consumers can now focus on creating app profiles and deploying their applications. You have essentially, at this point, enabled a Kubernetes PaaS experience for your organization.

Reviewer: ```suggestion
Alternatively, you could use App Mode to offer your own Palette-managed host clusters as a PaaS experience to downstream consumers. This concept is easier explained through an example. Assume you are a system administrator, and you want to expose Kubernetes to various in-house development teams. You could deploy several Kubernetes clusters to various platforms and create a " development " cluster group. You also ensured every cluster is enabled for Palette Virtual Cluster by selecting the option before deployment. You can now direct your organization members to use App Mode and create Palette Virtual Clusters as needed, or you can create virtual clusters ahead of time for them. The organization members or downstream consumers can now focus on creating app profiles and deploying their applications. You have essentially enabled a Kubernetes PaaS experience for your organization.
```

> This makes App mode unique from Cluster Mode, the ability to leverage an out-of-the-box host cluster group managed by us here at Spectro Cloud, or you can leverage Palette-managed Kubernetes clusters you already have deployed configured to your organization’s standards and expose them as Kubernetes Paas experience. As the consumer of app mode, you simply focus on deploying your application to a Kubernetes cluster by specifying the app profile. The overhead of managing infrastructure has essentially been removed for you, thus freeing up your time to focus on what matters the most, developing an application that solves business problems.   

Reviewer: This paragraph repeats and may not be needed. If you keep part or all of it, 'app mode' needs to be capitalized.

Author: I removed the first two sentences to reduce the repetition.

# How to access each mode? 

> You can quickly toggle between the two modes by navigating to your window's top right-hand side corner and expanding the user drop-down menu. Depending on which mode you are in, the name of the other mode will appear in the menu. Example: If you are in cluster mode, your menu will have the option “Switch to App Mode” and vice versa. 

Reviewer: ```suggestion
You can quickly toggle between App Mode and Cluster Mode by navigating to the **User** menu at top right and selecting the mode you want.
```

 

# App Mode or Cluster Mode?

> You might ask yourself, "How do I know which mode I should use?” The answer to that question comes down to your intent and objective. If you are enabling Kubernetes capabilities for others or configuring Palette, you will want to be in cluster mode. It’s in cluster mode, where you will find all the configuration options that a power user would want. If you are instead focused on consuming Kubernetes and want to simply deploy an application without worrying about the infrastructure overhead, then starting out with app mode is a good starting point. 

Reviewer: ```suggestion
You might ask yourself, "How do I know which mode I should use?” The answer comes down to your objective. 

- Choose Cluster Mode if you want to enable Kubernetes capabilities for others or configure Palette. Cluster Mode, provides all the configuration options a power user wants. 
- Choose App Mode if you want to simply deploy an application using Kubernetes without the infrastructure overhead. If you want to simply try out Palette, App Mode is a good starting point. 

```

> However, app mode might not meet your needs out of the box as the Spectro Cloud managed cluster group, beehive, imposes a resource limitation that could prevent an application from launching successfully if it is resource-heavy in nature. Review the resource quotas documentation to understand the limits of app mode better.  If you already have Palette-managed Kubernetes host clusters deployed or available to you in the form of a cluster group with Palette Virtual Mode enabled.  Then leveraging app mode is a great fit if you only want to focus on the developer experience aspect.

Reviewer: ```suggestion
App Mode may not meet your needs if your application requires a lot of resources. The Palette-managed cluster group, called Beehive, imposes a resource limitation that could prevent a resource-heavy application from launching successfully. Review the [Resource Quota](https://docs.spectrocloud.com/devx/resource-quota) documentation to understand App Mode limits.  If you already have Palette-managed Kubernetes host clusters deployed or available to you as a cluster group with Palette Virtual Clusters enabled, then leveraging App Mode is a great fit so you can focus on the developer experience aspect.
```
